core/src/main/java/org/apache/accumulo/core/client/rfile/RFileScanner.java

```diff
@@ -335,6 +352,52 @@ public boolean isSamplingEnabled() {
     public SamplerConfiguration getSamplerConfiguration() {
       return RFileScanner.this.getSamplerConfiguration();
     }
+
+    @Override
+    public TableId getTableId() {
+      return null;
```

Comment:
Returning null could cause an NPE in calling code that could be much later than this call and if the NPE is in a compound statement it could be hard to work back to the cause.  Thinking it would be better to fail fast rather than return null.

```suggestion
      throw new UnsupportedOperationException("This scanner is unrelated to any table or accumulo instance, it is operating directly on files."):
```


Alternatively we could always return a dummy table id here.

```suggestion
     return TableId.of("rfileScannerFakeD");
```

If we did that then the methods that take a table id could possibly only accept this id.

Reply:
Went with the latter, I assumed the way this would probably be used is `env.getPluginEnv().getConfiguration(env.getTableId())`. If we throw an error on `env.getTableId()` that might be confusing because not sure what user would pass to `getConfiguration(TableId)`... User could create their own dummy id but I feel it makes more sense for us to make it.